src/runtime/rpc/rpc_message.cpp

```diff
@@ -172,7 +172,7 @@ message_ex *message_ex::copy_message_no_reply(const message_ex &old_msg)
         static_cast<char *>(dsn::tls_trans_malloc(sizeof(message_header))),
         [](char *c) { dsn::tls_trans_free(c); });
     msg->header = reinterpret_cast<message_header *>(header_holder.get());
-    msg->header = {}; // initialize to empty struct
+    memset(static_cast<void *>(msg->header), 0, sizeof(message_header));
```

Comment:
I don't see the difference between the two statements. Please write a unit-test for `copy_message_no_reply`

Reply:
As we discuss offline, if header is an empty strtuct, it will lead problem in Line195, which is `msg->header->body_length = msg->buffers[1].length();`.